### PR TITLE
Integrate the diproche program

### PR DESCRIPTION
**Pull request by TimothyGillespie**

### CODE REVIEW

**Comment by TimothyGillespie** [comment]
The previous errors were due to window specific cmd commands. I made them run cross-platform now by using the shx library which allows to use UNIX commands on all platforms.

https://github.com/shelljs/shx

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@
 /node_modules
 /.pnp
 .pnp.js
+/src/util/diproche/diprocheProgram
```
Comment (Khazuar) [inline .gitignore]: I wouldn't put the diproche-program in the `src` folder if it's not actual source-code of this project.

File: src/util/diproche/diproche.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	private readonly session: PrologSession;
+
+	constructor() {
+		this.session = importFile("./diprocheProgram/diproche.pl");
+	}
+
+	public async getFeedBack(input: string): PrologResult {
+		const results: PrologResult | Promise<PrologResult> =  this.session.executeQuery(input);
+		return results;
+	}
```
Comment (Khazuar) [inline src/util/diproche/diproche.ts]: This is missing `Promise<..>` and `await`.

File: src/util/diproche/diproche.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+export default class Diproche {
+private readonly session: PrologSession;
+
+	constructor() {
+		this.session = importFile("../../.././readable_node_module_copies/diproche/diproche.pl");
+	}
+
+	public async getFeedback(input: string): Promise<ReadonlyMap<string, Array<string | boolean>>> {
+		const resultObject =  await this.session.executeQuery(input);
+		return new Promise(resolve => resolve(resultObject.getResults()));
```
Comment (Khazuar) [inline src/util/diproche/diproche.ts]: This whole method should read:
```typescript
return await this.session.executeQuery(input);
```

**Comment by TimothyGillespie** [comment]
Differences between SWI and tau-prolog forces us to pause this or find another solution since the current solution via a local SWI-Server is unsatisfactory.

**Pull request closed without merging**